include-refactoring/xsimd/arch/xsimd_arm7.hpp

```diff
@@ -0,0 +1,1942 @@
+                                                       float32x4_t>;
+
+            using excluding_int64_dispatcher = arm_dispatcher_impl<uint8x16_t, int8x16_t,
+                                                                   uint16x8_t, int16x8_t,
+                                                                   uint32x4_t, int32x4_t,
+                                                                   float32x4_t>;
+
+            /**************************
+             * comparison dispatchers *
+             **************************/
+
+            template <class T>
+            struct comp_return_type_impl;
+
+            template <>
+            struct comp_return_type_impl<uint8x16_t>
+            {
+                using type = uint8x16_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<int8x16_t>
+            {
+                using type = uint8x16_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<uint16x8_t>
+            {
+                using type = uint16x8_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<int16x8_t>
+            {
+                using type = uint16x8_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<uint32x4_t>
+            {
+                using type = uint32x4_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<int32x4_t>
+            {
+                using type = uint32x4_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<uint64x2_t>
+            {
+                using type = uint64x2_t;
+            };
+
+            template <>
+            struct comp_return_type_impl<int64x2_t>
+            {
+                using type = uint64x2_t;
+            };
+            
+            template <>
+            struct comp_return_type_impl<float32x4_t>
+            {
+                using type = uint32x4_t;
+            };
+
+            template <class T>
+            using comp_return_type = typename comp_return_type_impl<T>::type;
+
+            template <class... T>
+            struct arm_comp_dispatcher_impl : arm_dispatcher_base<comp_return_type, T...>
+            {
+            };
+
+            using excluding_int64_comp_dispatcher = arm_comp_dispatcher_impl<uint8x16_t, int8x16_t,
+                                                                             uint16x8_t, int16x8_t,
+                                                                             uint32x4_t, int32x4_t,
+                                                                             float32x4_t>;
+
+            /**************************************
+             * enabling / disabling metafunctions *
+             **************************************/
+
+            template <class T>
+            using enable_integral_t = typename std::enable_if<std::is_integral<T>::value, int>::type;
```

Comment:
out of curiosity: why `int` and not `void` in the `enable_if` ?

Reply:
Because in the case of SFINAE with functions, you may want to be able to enable / disable the functions for different conditions. The following would be illegal:

```cpp
template <class T, class = std::enable_if_t<std::is_integral<T>, value>>
void my_func1(T t) { ... }

template <class T, class = std::enable_if_t<std::is_floating_point<T>, value>>
void my_func(T t) { ...}
```

This is because default template parameters are not part of the function signature, therefore you are defining the same function twice.

The following, on the other hand, is totally fine:

```cpp
template <class T, std::enable_if_t<std::is_integral<T>::value, int> = 0>
void my_func(T t) {... }

template <class T, std::enable_if_t<std::is_floating_point<T>::value, int> = 0>
void my_func(T t) { ... }
```

Since the `enable_if` does not resolve to a default template parameter, but to a non type parameter when the condition is true.